### ISSUE TITLE
LIRC update [feature request]

### ISSUE THREAD

**Issue by kurac338** [osmc/osmc#451]
Hi, first of all, thank you very much for your great software.

I would like to comment about something.

Please if possible update the LIRC 0.9.0 package to 0.9.3 / 0.9.4

Currently USB IR Toys (specifically the IRDroid USB Transceiver) are not supported, since their driver is missing. I tried manually compiling and installing the packages but without success.

Or if possible instruct me of how to do so properly. I also tried to apt-get the new package but it collides with arm-v7-osmc-lirc (or something like that, I cannot remember of the top of my head).

Sorry if this is not the place to ask for such a thing. I will redirect the question to your forum.

Once again thanks for your support.

**Comment by samnazarko** [comment]
Unfortunately it's not possible for us to update LIRC at this time due to significant changes. It might be possible to backport support for the USB IR devices.

**Comment by kurac338** [comment]
I tried searching for a how to, but this seems out of my domain and knowledge. I could not even find how to add OSMC sources to download armv7-lirc-osmc and try to compile it with a newer lirc version. I might be approaching this wrong. Any help would be appreciated.

**Comment by mcurly** [comment]
@kurac338 and @samnazarko ,

Long time no see... well to make long story short, I've managed to get my dvb-t usb dongles aware of their infrared receivers by following the how-to [in this wiki](https://wiki.libreelec.tv/infrared_remotes) from libreelec. I had to make up for some differences in osmc, which (if my memory doesn't falter) are flagged in the libreelec how-to.

Hope you manage to sort your way @kurac338 .

Cheers!

**Comment by kurac338** [comment]
Hi, thanks for your comment, but reading through the link, it looks like an unrelated issue.

I managed to record the key presses from my IR remote without any problems.
I cannot get it to transmit the recorded signal. I keep getting the unsupported hardware error, which is true, since LIRC provided support for the USB transceiver since 0.9.3+.

I have also started considered just buying the IR LEDs any jump wiring them to the GPIO. Although that defeats the original purpose of having an all-in-one convenient dongle, but if that gets it to work without tinkering, hey, what can I do...

**Comment by mcurly** [comment]
@kurac338 , you're welcome.

I recall that when I first tried to set the infrared devices I was able to capture the keypresses too. But the weirdness happened when I tried to load and use the configuration file that I got from that scanning phase. There it was useless. It wasn't ever a task I could accomplish with LIRC itself (at least in OSMC distro). But I did manage to succeed with LIRC in raspberry but in Arch raspberry Pi's alarm distro.

So it is a bit disappointing to find out your problem isn't related. Nevertheless, and to rule out that awareness, I do invite you to give that how to a try.

Perhaps it may be profitable in the end.

Meanwhile, cheers and thanks for your time.

**Comment by samnazarko** [comment]
Hi chaps,

I hope you are well.

lsmod and dmesg from Arch where it works and OSMC where it doesn't could be interesting to see.

**Comment by mcurly** [comment]
@samnazarko thank you for that insightful request.


I am aware that the cloud of speculation is about to fall upon my head.

Since I don't have access to any of those logs, and trust is something nobody is willing to spend on the odd fellow, I'll just let it fall on me, and silence myself.

**Comment by samnazarko** [comment]
The reason I ask is I'd like to see if the LIRC bridge is being loaded on Arch.
It's possible that there is a kernel based decoder doing the work.

Our remote stack needs improvement, so I'm happy to work with others on it.

**Comment by kurac338** [comment]
I managed to get it working. For anyone in need of using an updated LIRC package (tested with v0.9.4d).

LIRC installation and setup process:
- Download the LIRC source code (https://sourceforge.net/projects/lirc/files/LIRC/)
- Unpack the archive, and follow the README. Basically call ./configure, check for any missing dependencies and install them. Then sudo make, and lastly sudo make install. Now the package will be installed in ./configure's default location (something like: /usr/local/bin/lirc/)
- Call "which" command on common LIRC executables (i.e. irrecord, irsend, lircd). Only lircd should now be pointing to the OSMC directory (something like: /usr/bin/lirc). Sym link the OSMC lircd to the newly created lircd.
- Reboot
- Check the versions of packages with -v. (lircd -v, should now be 0.9.4d, or whatever version was downloaded)

LIRC device specific notes:
Devices like mine, need to be explicitly noted when starting the lircd process. My start command now looks something like this: `sudo lircd -H <device driver name> -d </dev/ttyACM0 - the device path> -o <socket output directory - something like: /usr/var/run/lircd> <optional: full path to lircd.conf remote mapping file>`

**Comment by mcurly** [comment]
@kurac338 hello and I am glad you've managed to build latest lircd.

Have you already managed to get your remote working with it?

From what I understand you haven't, is that so?

And the procedure outlined in the tutorial from libreelec... have you given it a go?

Sincerely hope you manage to.

Meanwhile @samnazarko , I understand the specific need you have for that information and logs you were asking. I will try to find some time to reinstall Arch Linux arm and gather the desired info. I am sorry but other matters have kept me from being able to do so. I hope you understand. I also ask you @samnazarko if you are willing to advocate a Linux utility to make snapshots of Arch Linux arm systems. I mean both for backing up and restoring. Preferably with incremental capabilities. The reason why I am asking this is that it would definitely ease things on my side. I hope you understand. What do you use yourself? Why? How? Thank you.

**Comment by kurac338** [comment]
@mcurly I did manage to get it working; with the steps I described above. I have successfully recorded the remote control signals and I've sent them to my client device. So as far as the LIRC part is concerned, I am finished with it (for now). In the end I did not need to do the setup from your link.

If I find some spare time, I will try to update my previous post today, and also try to figure out where the LIRC process is started on OSMC boot; so that I can use my own specific parameters on system startup.

Thanks for the support!
